src/backend/utils/misc/guc_gp.c

```diff
@@ -5177,6 +5177,16 @@ check_optimizer(bool *newval, void **extra, GucSource source)
 			GUC_check_errmsg("External planner is not registered");
 			return false;
 		}
+		else if (NULL == planner_hook)
+			/*
+			 * But in case GUC source <= PGC_S_ARGV, this function may be called
+			 * before external planner is loaded from a shared lib. For ex.,
+			 * user may set the GUC in 'postgresql.conf', so system will try to
+			 * apply the GUC before the init of shared libs. We need to allow
+			 * such cases, assuming that the user knows what he is doing.
+			 */
+			elog(LOG, "'optimizer' is explicitly set to 'on'. "
```

Comment:
maybe warning?

Reply:
I thought about it.

I think that warning is considered to be smth that is better to fix. But there will be no way the user can fix this warning with current Postgres GUC architecture. So it can uncomfortable for a pedantic user.

I wanted to have a kind of a hint for the user, thus I chose LOG instead of WARNING.